src/lib/core/CHIPCallbackManager.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *  @file
+ *    This file contains definitions for Callback objects for registering with
+ *     Clusters and the Device
+ */
+
+#pragma once
+
+#include "CHIPCallbackManager.h"
+#include <stddef.h>
+#include <stdint.h>
+
+using namespace chip;
+using namespace Callback;
+
+void CallbackManager::AddPostAttributeChangeHandler(Callback::PostAttributeChangeCallback<> * onResponse, EndpointId endpointId,
+                                                    ClusterId clusterId, AttributeId attributeId)
+{
+    // Since a cb is binded to endpoint and other args, should we have the init take those arguments to
+    // make that relationship explicit. Or should we bind somewhere else?
+    onResponse->mAttributeID = attributeId;
+    onResponse->mEndpointID  = endpointId;
+    onResponse->mClusterID   = clusterId;
+
+    Callback::Cancelable * cancelable = onResponse->Cancel();
+    mCallbacks.Enqueue(cancelable);
```

Comment:
you need to set the cancel function in the cancelable so that you can dequeue this callback on destruction